Adding Swagger for POST APIs used to upload a customer TDE certificates

nivimsft opened the pull request:
…e in CMS

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [X] The title of the PR is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [X] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [X] If applicable, the PR references the bug/issue that it fixes.
- [X] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [X] My spec meets the review criteria:
  - [X] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [X] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

nivimsft committed:
Adding Swagger for POST APIs used to upload a customer TDE certificate in CMS

AutorestCI commented:
# Automation for azure-libraries-for-java

AutorestCI commented:
# Automation for azure-sdk-for-python

AutorestCI commented:
# Automation for azure-sdk-for-ruby

AutorestCI commented:
# Automation for azure-sdk-for-go

jaredmoo commented on specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/ManagedInstanceTdeCertificates.json:
```diff
@@ -0,0 +1,245 @@
+        }
+      }
+    },
+    "Resource": {
```
Please manually remove Resource & ProxyResource and edit TdeCertificate to reference `"../../../common/v1/types.json#/definitions/ProxyResource"` instead

nivimsft replied:
Thanks for review, I have addressed this comment in the latest commit.

jaredmoo commented on specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/ManagedInstanceTdeCertificates.json:
```diff
@@ -0,0 +1,245 @@
+      "type": "object",
+      "properties": {
+        "certificateName": {
+          "description": "The name of the certificate",
```
This property is readonly but the API is write-only. So this property is basically unused. Can you remove it?

nivimsft replied:
Thanks for review, I have addressed this comment in the latest commit.

jaredmoo reviewed (changes requested):
Mostly looks good, minor comments

nivimsft committed:
Addressing Jared's comment on PR
- Remove certificateName property
- Remove Resource and ProxyResource manually
- Edit TdeCertificate to reference "../../../common/v1/types.json#/definitions/ProxyResource"

jaredmoo commented:
Please fix validator issues: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/392836233

nivimsft committed:
Addressing validation example errors

jaredmoo commented:
There is still model validation error: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/392880069

nivimsft committed:
Addressing validation example errors

jaredmoo commented:
@nivimsft this is good to merge now right?

marstr reviewed (approved):
Naming these Operation Groups "TdeCertificate_Create" and "ManagedInsanceTdeCertificate_Create" will force users to create a new instance of a client and give it an authorizer. If possible, I'd recommend tying these operations to broader clients, something like "Certificates_CreateTdeCertificate". However, I'll leave this up to your and @jaredmoo's discretion.

marstr commented:
Still waiting on ARM, only a suggestion above. :)

jaredmoo commented:
Understood, but trying to be clever with OperationId's led us to a very awkward place for some APIs (see https://www.nuget.org/packages/Microsoft.Azure.Management.Sql/1.7.0-preview where I made several breaking changes to unravel this problem). To preserve long-term sanity of the basic (non-fluent) SDKs I am making it a point to strictly have a separate operations class per resource type.

jaredmoo commented:
In this particular example it is entirely possible to implement GET/DELETE in the future, if we do then we will be very happy to have separate operations classes for db versus managed instance certificates. Since there are many teams in SQL independently making new APIs, it is much clearer for everyone in the long run if we have strict and simple rules for OperationId's.

marstr commented:
You make me a very happy man, @jaredmoo. Thanks for the explanation. Consider the caveat on my approval removed.

jaredmoo commented:
@ravbhatnagar please provide ARM feedback :) thx

nivimsft committed:
Adding to all package-composite-v* and package-pure of appropriate version

AutorestCI commented:
# Automation for azure-sdk-for-java
